### PR TITLE
libobs: Don't monitor hidden sources on macOS

### LINKED ISSUE #6670: Audio mixer : Hide Media source entry when audio still play
### Operating System Info

macOS 11

### Other OS

_No response_

### OBS Studio Version

27.2.4

### OBS Studio Version (Other)

_No response_

### OBS Studio Log URL

https://obsproject.com/logs/1foqr438zlNQkZ6r

### OBS Studio Crash Log URL

_No response_

### Expected Behavior

Audio mixer entry for hidden "media source" with audio playing stay visible.

### Current Behavior

When hiding a media source with audio (and when "restart playback when source become active" unchecked), audio continue to play, but audio mixer entry disappear.

### Steps to Reproduce

1. Import files with Audio using Media source
2. Uncheck "Restart playback when sources become visble"
3. Hide the layer
4. Audio mixer entry disappear, but audio still play

### Anything else we should know?

_No response_

### PR DESCRIPTION
**Pull request by tuduweb**
### Description

Add judgement to check whether source is active in audio monitoring like other backends.

https://github.com/obsproject/obs-studio/blob/f6f6690ccfeb4b78eec82d4ff853cb6db7138860/libobs/audio-monitoring/pulse/pulseaudio-output.c#L230-L231

https://github.com/obsproject/obs-studio/blob/f6f6690ccfeb4b78eec82d4ff853cb6db7138860/libobs/audio-monitoring/win32/wasapi-output.c#L316-L318

### Motivation and Context

Fixes #6670

### How Has This Been Tested?

Test in MacOS 11.2 M1
Reproduce those steps, audio muted when Media Source hidden.

### Types of changes

Bug fix (non-breaking change which fixes an issue) 

### Checklist:

- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.
- [x] I have included updates to all appropriate documentation.

### CODE REVIEW

**Review by PatTheMav — APPROVED** [review]
Looks good to me.

**Comment by PatTheMav** [comment]
Minor nitpick - please rename PR and commit to more precisely describe the change, e.g.

> libobs: Don't monitor hidden sources on macOS

File: libobs/audio-monitoring/osx/coreaudio-output.c [hunk 1]
```diff
@@ -65,6 +65,10 @@ static void on_audio_playback(void *param, obs_source_t *source,
 		return;
 	}
 
+	if (os_atomic_load_long(&source->activate_refs) == 0) {
+		return;
+	}
```
Comment (norihiro) [inline libobs/audio-monitoring/osx/coreaudio-output.c:70]:
```suggestion
	if (os_atomic_load_long(&source->activate_refs) == 0)
		return;
```

**Comment by tuduweb** [comment]
> Minor nitpick - please rename PR and commit to more precisely describe the change, e.g.
> 
> > libobs: Don't monitor hidden sources on macOS

thanks for your advice and revise, I will pay more attention next time.

**Pull request merged**